### PR TITLE
14871 add testrunner

### PR DESCRIPTION
**Pull request by tsmaeder**
### What does this PR do?

Adds the Java testrunner extension to all Java-related plugins. The version is 0.24.0, since 0.24.1 requires Java 11 to run.

This issue fixes Installing Java Test Runner Support for Java https://github.com/eclipse/che/issues/14871

### CODE REVIEW

File: v3/plugins/redhat/java8/0.63.0a/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+apiVersion: v2
+publisher: redhat
+name: java
```
Comment (ericwill) [inline v3/plugins/redhat/java8/0.63.0a/meta.yaml:3]: Should be `java8`

**Review by svor — APPROVED** [review]
LGTM

**Comment by ericwill** [comment]
We should probably deprecate `0.63.0` and set it to migrate to `0.63.0a`

**Comment by rhopp** [comment]
[ci-test-happy-path]

**Comment by rhopp** [comment]
There was probably some glitch... I couldn't reproduce this running the tests locally and  next run of the PR check succeeded.

**Comment by tsmaeder** [comment]
> 
> 
> We should probably deprecate `0.63.0` and set it to migrate to `0.63.0a`

Hhmh...why? If anything, shouldn't we deprecate 0.57? 0.63 works just fine, and if the user wants the new features, he can update by hand. There is nothing special about 0.63a except that the version of vscode-java did not change.

**Comment by tsmaeder** [comment]
[ci-test]

**Comment by ericwill** [comment]
> > We should probably deprecate `0.63.0` and set it to migrate to `0.63.0a`
> 
> Hhmh...why? If anything, shouldn't we deprecate 0.57? 0.63 works just fine, and if the user wants the new features, he can update by hand. There is nothing special about 0.63a except that the version of vscode-java did not change.

I suppose it could be optional, but it would be nice to have all users be able to take advantage of the new test runner support out of the box, and it's not immediately obvious that such new features are available just by visual inspection. Since there is no `vscode-java` version change from `0.63.0` to `0.63.0a` it's a harmless migration.

File: v3/plugins/redhat/java11/0.63.0a/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      name: vscode-java
+      memoryLimit: "1500Mi"
+      volumes:
+      - mountPath: "/home/theia/.m2"
+        name: m2
+  extensions:
+    - https://download.jboss.org/jbosstools/vscode/3rdparty/vscode-java-debug/vscode-java-debug-0.26.0.vsix
+    - https://download.jboss.org/jbosstools/static/jdt.ls/stable/java-0.63.0-2222.vsix
+    - https://open-vsx.org/api/vscjava/vscode-java-test/0.24.0/file/vscjava.vscode-java-test-0.24.0.vsix
+
```
Comment (ericwill) [inline v3/plugins/redhat/java11/0.63.0a/meta.yaml:25]: Extra newlines/missing newling at EOF? Looks like something weird happened here.

**Comment by tsmaeder** [comment]
[ci-test-happy-path]

**Comment by tsmaeder** [comment]
[ci-test-happy-path]

**Comment by benoitf** [comment]
I've rebased this PR

**Comment by ericwill** [comment]
Superseded by #619

**Pull request closed without merging**